src/openApi/v3/parser/getMappedType.ts

```diff
@@ -1,32 +1,45 @@
-const TYPE_MAPPINGS = new Map<string, string>([
-    ['file', 'binary'],
-    ['any', 'any'],
-    ['object', 'any'],
-    ['array', 'any[]'],
-    ['boolean', 'boolean'],
-    ['byte', 'number'],
-    ['int', 'number'],
-    ['integer', 'number'],
-    ['float', 'number'],
-    ['double', 'number'],
-    ['short', 'number'],
-    ['long', 'number'],
-    ['number', 'number'],
-    ['char', 'string'],
-    ['date', 'string'],
-    ['date-time', 'string'],
-    ['password', 'string'],
-    ['string', 'string'],
-    ['void', 'void'],
-    ['null', 'null'],
+interface MappedType {
+    type: string;
+    isPrimitive: boolean;
+}
+
+const TYPE_MAPPINGS = new Map<string, MappedType>([
+    ['file', { type: 'binary', isPrimitive: true }],
+    ['any', { type: 'any', isPrimitive: true }],
+    ['object', { type: 'any', isPrimitive: true }],
+    ['array', { type: 'any[]', isPrimitive: true }],
+    ['boolean', { type: 'boolean', isPrimitive: true }],
+    ['byte', { type: 'number', isPrimitive: true }],
+    ['int', { type: 'number', isPrimitive: true }],
+    ['integer', { type: 'number', isPrimitive: true }],
+    ['float', { type: 'number', isPrimitive: true }],
+    ['double', { type: 'number', isPrimitive: true }],
+    ['short', { type: 'number', isPrimitive: true }],
+    ['long', { type: 'number', isPrimitive: true }],
+    ['number', { type: 'number', isPrimitive: true }],
+    ['char', { type: 'string', isPrimitive: true }],
+    ['date', { type: 'string', isPrimitive: true }],
+    ['date-time', { type: 'string', isPrimitive: true }],
+    ['password', { type: 'string', isPrimitive: true }],
+    ['string', { type: 'string', isPrimitive: true }],
+    ['void', { type: 'void', isPrimitive: true }],
+    ['null', { type: 'null', isPrimitive: true }],
+]);
+
+const FORMAT_MAPPINGS = new Map<string, MappedType>([
+    ['binary', { type: 'binary', isPrimitive: true }],
+    ['float', { type: 'number', isPrimitive: true }],
+    ['date', { type: 'string', isPrimitive: true }],
+    ['date-time', { type: 'string', isPrimitive: true }],
+    ['Big', { type: 'Big', isPrimitive: false }],
```

Comment:
I don't think we want to use `Big` for the format here? It's TS/JS-specific and we can't put it in the OpenAPI schema. `decimal` would be more appropriate I think

Reply:
Agree with the `decimal` wording for sure. As stated before, this was mostly to test. I've removed the introduction of actual custom types since these need to be discussed 👍